src/sagemaker/workflow/emr_step.py

```diff
@@ -0,0 +1,119 @@
+from sagemaker.workflow.entities import (
+    RequestType,
+)
+from sagemaker.workflow.properties import (
+    Properties,
+)
+from sagemaker.workflow.steps import Step, StepTypeEnum, CacheConfig
+
+
+class EMRStepConfig:
+    """Config for a Hadoop Jar step."""
+
+    def __init__(
+        self, jar, args: List[str] = None, main_class: str = None, properties: List[dict] = None
+    ):
+        """Create a definition for input data used by an EMR cluster(job flow) step.
+
+        See AWS documentation on the ``StepConfig`` API for more details on the parameters.
+
+        Args:
+            args(List[str]):
+                A list of command line arguments passed to
+                the JAR file's main function when executed.
+            jar(str): A path to a JAR file run during the step.
+            main_class(str): The name of the main class in the specified Java file.
+            properties(List(dict)): A list of key-value pairs that are set when the step runs.
+        """
+        self.jar = jar
+        self.args = args
+        self.main_class = main_class
+        self.properties = properties
+
+    def to_request(self) -> RequestType:
+        """Convert EMRStepConfig object to request dict."""
+        config = {"HadoopJarStep": {"Jar": self.jar}}
+        if self.args is not None:
+            config["HadoopJarStep"]["Args"] = self.args
+        if self.main_class is not None:
+            config["HadoopJarStep"]["MainClass"] = self.main_class
+        if self.properties is not None:
+            config["HadoopJarStep"]["Properties"] = self.properties
+
+        return config
+
+
+class EMRStep(Step):
+    """EMR step for workflow."""
+
+    def __init__(
+        self,
+        name: str,
+        display_name: str,
+        description: str,
+        job_flow_id: str,
+        step_config: EMRStepConfig,
+        depends_on: List[str] = None,
+        cache_config: CacheConfig = None,
+    ):
+        """Constructs a LambdaStep.
+
+        Args:
+            name(str): The name of the EMR step.
+            display_name(str): The display name of the EMR step.
+            description(str): The description of the EMR step.
+            job_flow_id(str): A string that uniquely identifies the job flow(cluster).
+            step_config(EMRStepConfig): One StepConfig to be executed by the job flow.
+            depends_on(List[str]):
+                A list of step names this `sagemaker.workflow.steps.EMRStep` depends on
+            cache_config(CacheConfig):  A `sagemaker.workflow.steps.CacheConfig` instance.
+
+        """
+        super(EMRStep, self).__init__(name, display_name, description, StepTypeEnum.EMR, depends_on)
+
+        emr_step_args = {"JobFlowId": job_flow_id, "StepConfig": step_config.to_request()}
+        self.args = emr_step_args
+        self.cache_config = cache_config
+
+        self._properties = Properties(
+            path=f"Steps.{name}", external_service_name="emr", shape_name="DescribeStepOutput"
```

Comment:
Shape is wrong (sorry, doc is probably outdated, take a look at the EMRStepProperties in the pipeline definition package).